### DISCUSSION TITLE
REST API is documented here https://github.com/minio/minio/tree/master/pkg/madmin , for now, we are not planning to implement different language libraries for Admin REST API.  Our SDKs are only meant for S3 APIs only.

### DISCUSSION THREAD

**Discussion by awezmapr** [minio/minio#20218] [Q&A] [answered]
Link is not available.

              REST API is documented here https://github.com/minio/minio/tree/master/pkg/madmin , for now, we are not planning to implement different language libraries for Admin REST API.  Our SDKs are only meant for S3 APIs only. 

The suggested way is to use `mc` tool or use the Go API directly or write your own wrapper for your language choice.

_Originally posted by @harshavardhana in https://github.com/minio/minio/issues/7539#issuecomment-482898917_

**Comment by klauspost** [comment, marked as answer]
https://github.com/minio/madmin-go

**Reply by awezmapr** [reply]
Thanks for the **Go API** documentation.
My understanding is REST API is not supported/documentation is no longer available.

**Reply by klauspost** [reply]
> My understanding is REST API is not supported/documentation is no longer available.

Correct. We do not document beyond what is on madmin-go.

**Comment by decoursin** [comment]
Yeah, I would also very much appreciate documentation of the Admin REST API. I'm surprised this isn't available. I would have imagined that I should be able to at least query my minio server to find out the health status, admin info, etc.

**Reply by decoursin** [reply]
This is the best I could find, but apparently it's way outdated: https://gist.github.com/krishnasrinivas/ea5603d16e6d706793332af674df83e3

**Reply by klauspost** [reply]
https://min.io/docs/minio/kubernetes/upstream/operations/monitoring/healthcheck-probe.html provides health checks.

**Reply by klauspost** [reply]
@krisis ?

**Reply by harshavardhana** [reply]
@decoursin https://pkg.go.dev/github.com/minio/madmin-go/v3

**Reply by decoursin** [reply]
@klauspost thanks for sending me the health check API. I didn't know about it.

@harshavardhana it would be nice if it was documented at least _how_ the go client communicates with the server.
